gobblin-utility/src/main/java/gobblin/util/request_allocation/AllEqualAllocator.java

```diff
@@ -0,0 +1,66 @@
+package gobblin.util.request_allocation;
+
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.Iterator;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Iterators;
+import com.typesafe.config.Config;
+
+import javax.annotation.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * A {@link RequestAllocator} optimized for the case when all work units have the same priority. Note the input {@link Comparator}
+ * will be ignored.
+ */
+@Slf4j
+@AllArgsConstructor
+public class AllEqualAllocator<T extends Request<T>> implements RequestAllocator<T> {
+
+  public static class Factory implements RequestAllocator.Factory {
+    @Override
+    public <T extends Request<T>> RequestAllocator<T> createRequestAllocator(Comparator<T> prioritizer,
+        ResourceEstimator<T> resourceEstimator, Config limitedScopeConfig) {
+      return new AllEqualAllocator<>(resourceEstimator);
+    }
+  }
+
+  private final ResourceEstimator<T> resourceEstimator;
+
+  @Override
+  public AllocatedRequests<T> allocateRequests(Iterator<? extends Requestor<T>> requestors, ResourcePool resourcePool) {
+    ConcurrentBoundedPriorityIterable<T> iterable =
+        new ConcurrentBoundedPriorityIterable<>(new AllEqualComparator(), this.resourceEstimator, resourcePool);
+
+    Iterator<T> unionIterator = Iterators.concat(Iterators.transform(requestors, new Function<Requestor<T>, Iterator<T>>() {
```

Comment:
This does not seem `AllEqual`. Requestors that are in front of others always get more requests to be allocated.